Feat/product model

skarltjr opened the pull request:
기본 product aggregate 
- vo를 활용하여 완전한 하나의 개념 표현
- ex) product의 정보를 productInfo라는 vo로 표현하여 모든 상품이 공통적으로 갖는 하나의 정보를 표현
- 상품 : 카테고리 다대다 풀기

skarltjr committed:
feat : basic product aggregate

skarltjr committed:
feat : basic product aggregate

skarltjr commented:
@ktj1997   더 필요한 필드있는지 계속 생각해보져

rlaxowns7916 commented on product-service/src/main/java/com/msa/domain/vo/ProductInfo.java:
```diff
@@ -0,0 +1,21 @@
+    private Money price;
+
+    @Column(name = "product_stock",nullable = false)
+    private int stock;
```
Stock도 Money처럼 VO로만드는건 어떨까요
Stock도 음수로 내려간다거나 생성할 때 이런 Validation이나 로직을 가지려면 VO로 뺴는게좋을 것 같 아서요

skarltjr replied:
오 stock도 고민해봤는데 stock을 vo로 가져갔을 때 변수명을 여기 변수명을 어떻게 가져갈까 고민임돠 
private Stock stock도 상관이 없을지 아니면 더 자세히 표현해줄만한 변수명이 없을지 생각해봐야겠네여 굳굳

skarltjr committed:
fix : product stock with using vo
- 상품의 재고는 Stock이라는 하나의 vo 다룬다
- 재고수를 하나의 Stock이라는 개념으로 표현한다.
- 추가로 별도의 validation을 수행할 수 있도록한다.

skarltjr committed:
chore : add AllArgsConstructor
